pkg/endpoint/bpf.go

```diff
@@ -940,20 +941,57 @@ func (e *Endpoint) e.addPolicyKey(keyToAdd policy.Key, entry policy.MapStateEntr
 		e.getLogger().WithError(err).Errorf("Failed to add PolicyMap key %s %d", policyKeyToPolicyMapKey.String(), entry.ProxyPort)
 		return err
 	}
-	
+
 	// Operation was successful, add to realized state.
 	e.realizedPolicy.PolicyMapState[keyToAdd] = entry
 
 	return nil
 }
 
+func (e *Endpoint) proxyID(key policy.Key) string {
+	return policy.ProxyID(e.ID, key.TrafficDirection == trafficdirection.Ingress.Uint8(), u8proto.U8proto(key.Nexthdr).String(), key.DestPort)
+}
+
+func (e *Endpoint) applyPolicyMapChanges() error {
+	errors := []error{}
+
+	adds, deletes := e.realizedPolicy.PolicyMapChanges.GetMapChanges()
+
+	e.getLogger().Debugf("applyPolicyMapChanges: adds: %v, deletes: %v", adds, deletes)
+
+	for keyToAdd, entry := range adds {
+		// Keep the existing proxy port, if any
+		entry.ProxyPort = e.realizedRedirects[e.proxyID(keyToAdd)]
+		err := e.addPolicyKey(keyToAdd, entry)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	for keyToDelete := range deletes {
+		err := e.deletePolicyKey(keyToDelete)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	if len(errors) > 0 {
+		return fmt.Errorf("updating desired PolicyMap state failed: %s", errors)
+	} else if len(adds)+len(deletes) > 0 {
+		e.getLogger().Infof("Applied policy updates due to added identities: %v, deleted identities: %v", adds, deletes)
```

Comment:
Same here. Also, I don't think this needs to be `Infof`

Reply:
Changed to debug with logfields.